Changes to volume/HU/init slice selection for SPR abstract

jmweaver-FDA opened the pull request:
Summary of changes:
- CT_dataset_pipeline.py: loading .csv files containing volume and HU distributions from BHSD dataset. Distribution is sampled to generate synthetic lesions. I also opted for looping through the requested number of cases and randomly applying lesion, patient, and acquisition parameters instead of constructing a large vector of all possible parameter combinations and shuffling them. I've left the old method commented out and if we want to return to it we can discuss.
- Phantoms.py: Consistency between spheres, epidural, and subdural: "contrast" changed to "intensity" - argument to lesion insertion is now the intensity (HU) of the lesion, not contrast relative to white matter. I've also changed "contrast" to "intensity" in most other variable names.
- Lesion_definition.py: Initial starting slice for epidural and subdural lesions is now only sampled from the top third of the skull. This can be changed in the future. For now, it prevents large hemorrhages from being initiated on slices in which there is no room to grow (very inferior skull slices)
- Image_acquisition.py: Added call to XCIST function to obtain new mu value if monochromatic energy is specified.

Also of interest is RSNA_BHSD_fitting.ipynb, which has been updated to subdivide individual hemorrhages into clusters to be used for more detailed analysis. For now, because we're generating single lesions, it prevents us from generating abnormally large hemorrhages (For example, a patient with a 200 mL epidural hemorrhage mask may actually have multiple epidural hemorrhages, not one single large hemorrhage).

jmweaver-FDA committed:
big changes to volume and HU selection

jmweaver-FDA committed:
updates to my personal recon testing jupyter notebook

jmweaver-FDA committed:
Merge branch 'master' into jayse_dev

jmweaver-FDA committed:
updates to init slice random selection

Commit:
added results with bhc correction

jmweaver-FDA committed:
updated BHSD fitting to subdivide lesions into clusters

jmweaver-FDA committed:
mu change to switch between mono and polychromatic

brandonjnelsonFDA committed:
generalized sphere lesion to ellipse

brandonjnelsonFDA committed:
replaced custom ellipsoid with scikit-image implementation

brandonjnelsonFDA committed:
fix to new ellipse

brandonjnelsonFDA committed:
added distance transform so only points where the lesion could fit are sampled

brandonjnelsonFDA committed:
with distance transform, removed iterative search, since every point is valid for sphere

brandonjnelsonFDA committed:
removed reference to tol which is no longer used since adding distance map for sphere lesion insertion

brandonjnelsonFDA committed:
added edema and eccentric round lesions, changed name to round

Commit:
figures for research seminar

Commit:
Merge branch 'brandon_dev' of https://github.com/bnel1201/PedSilicoICH into brandon_dev

jmweaver-FDA committed:
fix lesion mask output bug

brandonjnelsonFDA commented on CT_dataset_pipeline.py:
Summary bullet 1: I like replacing the multiple nested for loops for iterating parameters with individual random choice on lines 73-98, and I like the option to include a csv file for distributions, this is similar to VICTRE pipeline and xcist for having parameter files. We can strive for consistent format/style with them

Commit:
precommit before exploring shueyues segmentation augmentations

Commit:
renamed all contrast -> intensity

Commit:
trying to merge with PRs together

Merge branch 'brandon_dev' of https://github.com/bnel1201/PedSilicoICH into jayse_dev_exp

brandonjnelsonFDA commented on src/pedsilicoICH/ground_truth_definition/phantoms.py:
```diff
@@ -20,7 +20,10 @@
+    '''
+    Converts volume in mL to radii in mm
+    '''
+    return np.power(3/4/np.pi*volume*1000, 1/3)
```
previously volume was in voxels (unitless), now that its in mL (which is good) we need to update all of the default parameters for volume which previously were in voxels, e.g. here where the default was 200 voxels, but now 200 mL is way too big: https://github.com/brandonjnelsonFDA/PedSilicoICH/blob/4fa3d015d0e3e6199bb9d98fa04dcf69260e6032/src/pedsilicoICH/ground_truth_definition/phantoms.py#L236

brandonjnelsonFDA commented on src/pedsilicoICH/lesion_definition.py:
```diff
@@ -55,26 +66,33 @@ def insert_dural_3D(phantom, init_slice, hematoma_type, mass_effect,
+                    count += 1
+                    init_slice = int(random.choice(np.linspace(0, int(HU_array.shape[0]/3), int(HU_array.shape[0]/3) + 1)))
+                else:
+                    count = tol
```
it would be useful to add in an Error message if this while loop is failing due to requesting too large of a volume. As currently implemented a try-except clause means that any errors are hidden which can make it hard for a user to realize they are requesting too large of a volume.

Commit:
merge jayses commits with brandons for PR

Commit:
fixed test with new mL volumes

Commit:
commented out experimental line for mass effect in round lesions
